### ISSUE TITLE
Using the image make our engine slow by 8 times

### ISSUE THREAD

**Issue by alphamarket** [nginx-proxy/acme-companion#907]
When I turn on the `LETSENCRYPT_HOST` on my image it makes the connections to my engine almost 8 times slower, here is the configuration that I have set up:

```yaml
version: "3.9"

services:
  tracker:
    image: username/image
    deploy:
      replicas: 3
      update_config:
        parallelism: 1
        delay: 30s
        order: start-first
    ports:
      - "3000"
      - "9100:3000"
    environment:
      - VIRTUAL_HOST=domain.com
      - VIRTUAL_PORT=3000

      - LETSENCRYPT_HOST=domain.com

  nginx-proxy:
    image: jwilder/nginx-proxy
    ports:
      - "80:80"
      - "443:443"
    labels:
      com.github.jrcs.letsencrypt_nginx_proxy_companion.nginx_proxy: "true"
    volumes:
      - certs:/etc/nginx/certs:ro
      - vhostd:/etc/nginx/vhost.d
      - html:/usr/share/nginx/html
      - /var/run/docker.sock:/tmp/docker.sock:ro

  letsencrypt:
    image: nginxproxy/acme-companion
    depends_on:
      - nginx-proxy
    environment:
      - NGINX_PROXY_CONTAINER=nginx-proxy
      - DEFAULT_EMAIL=name@email.com
    volumes:
      - acme:/etc/acme.sh
      - certs:/etc/nginx/certs
      - vhostd:/etc/nginx/vhost.d
      - html:/usr/share/nginx/html
      - /var/run/docker.sock:/var/run/docker.sock:ro

volumes:
  acme:
  certs:
  html:
  vhostd:
```
 With `LETSENCRYPT_HOST` enabled on the image our benchmark is as follow:

```
ab -n 1000 -c 200 'https://domain.com'
This is ApacheBench, Version 2.3 <$Revision: 1843412 $>
Copyright 1996 Adam Twiss, Zeus Technology Ltd, http://www.zeustech.net/
Licensed to The Apache Software Foundation, http://www.apache.org/

Benchmarking domain.com (be patient)
Completed 100 requests
Completed 200 requests
Completed 300 requests
Completed 400 requests
Completed 500 requests
Completed 600 requests
Completed 700 requests
Completed 800 requests
Completed 900 requests
Completed 1000 requests
Finished 1000 requests


Server Software:        nginx/1.21.4
Server Hostname:        domain.com
Server Port:            443
SSL/TLS Protocol:       TLSv1.2,ECDHE-RSA-AES256-GCM-SHA384,4096,256
Server Temp Key:        X25519 253 bits
TLS Server Name:        domain.com

Document Path:          /
Document Length:        0 bytes

Concurrency Level:      200
Time taken for tests:   4.058 seconds
Complete requests:      1000
Failed requests:        0
Non-2xx responses:      1000
Total transferred:      289000 bytes
HTML transferred:       0 bytes
Requests per second:    246.43 [#/sec] (mean)
Time per request:       811.582 [ms] (mean)
Time per request:       4.058 [ms] (mean, across all concurrent requests)
Transfer rate:          69.55 [Kbytes/sec] received

Connection Times (ms)
              min  mean[+/-sd] median   max
Connect:       10  509 987.6     60    2751
Processing:     3  199 354.7     64    1301
Waiting:        2  199 354.8     64    1301
Total:         21  708 1299.6    125    4035

Percentage of the requests served within a certain time (ms)
  50%    125
  66%    161
  75%    201
  80%    268
  90%   3789
  95%   3842
  98%   3864
  99%   3875
 100%   4035 (longest request)
```

But when we comment the `LETSENCRYPT_HOST` in the image we get the following benchmark:

```
ab -n 10000 -c 200 'http://domain.com'
This is ApacheBench, Version 2.3 <$Revision: 1843412 $>
Copyright 1996 Adam Twiss, Zeus Technology Ltd, http://www.zeustech.net/
Licensed to The Apache Software Foundation, http://www.apache.org/

Benchmarking domain.com (be patient)
Completed 1000 requests
Completed 2000 requests
Completed 3000 requests
Completed 4000 requests
Completed 5000 requests
Completed 6000 requests
Completed 7000 requests
Completed 8000 requests
Completed 9000 requests
Completed 10000 requests
Finished 10000 requests


Server Software:        
Server Hostname:        domain.com
Server Port:            80

Document Path:          /
Document Length:        0 bytes

Concurrency Level:      200
Time taken for tests:   4.895 seconds
Complete requests:      10000
Failed requests:        0
Non-2xx responses:      10000
Total transferred:      1960000 bytes
HTML transferred:       0 bytes
Requests per second:    2042.72 [#/sec] (mean)
Time per request:       97.909 [ms] (mean)
Time per request:       0.490 [ms] (mean, across all concurrent requests)
Transfer rate:          390.99 [Kbytes/sec] received

Connection Times (ms)
              min  mean[+/-sd] median   max
Connect:        0    4  62.0      0    1037
Processing:     1   91  65.3     77     298
Waiting:        1   91  65.3     77     298
Total:          1   96  88.6     77    1152

Percentage of the requests served within a certain time (ms)
  50%     77
  66%    106
  75%    153
  80%    167
  90%    189
  95%    209
  98%    227
  99%    239
 100%   1152 (longest request)
```

TL;DR When we use the `LETSENCRYPT_HOST` the RPS (Request per second) is `246.43` but when we comment it; it's `2042.72` which is almost 8 times bigger than when we don't use the `LETSENCRYPT_HOST`.

**Question:** Why is this happening, and can we boost up the speed?

**Comment by buchdag** [comment]
I don't really think there is an issues there, that's expected with TLS v1.2, see https://serverfault.com/a/934309

Bottom line is : 
- the TLS handshake has a significant added cost when doing benchmarks with `ab`
- TLSv1.3 should be better in this regard but `ab` does not support TLSv1.3 at the moment
- `ab` testing is not representative of real world usage

You can see in your own benchmark that the `connect` time goes from 4s on `HTTP` to 500s on `HTTPS`.  Testing on `google.com` does not yeld significantly different number, with the mean `connect` time being 5 times higher for `HTTPS`.

By the way nginxproxy/acme-companion is not responsible for proxy settings that could influence this, [nginxproxy/nginx-proxy](https://github.com/nginx-proxy/nginx-proxy) is. Do not hesitate to open a discussion there to see if we have room for improvements.

**Comment by buchdag** [comment]
Inactive issue, closing.
